tests/utils.py

```diff
@@ -256,7 +256,7 @@ async def compile_done():
         code = compiling.code
         return code == 204
 
-    await retry_limited(compile_done, 10)
+    await retry_limited(compile_done, 20)
```

Comment:
@sanderr @arnaudsjs 
Without this `tests/server/test_compilerservice.py::test_server_recompile` fails. So I likely slowed something down a lot but I don't know what. I added 2 flamegraphs to the pr (one of the test run on master, the other off the test run on this branch) but I didn't help me a lot.
So I think I could use some help finding out what I did wrong.

Reply:
I will have a look.

Reply:
It seems that the `pip install` command, that installs the dependencies of the std module in the compiler venv, has become a lot slower. It think that's quite normal, given the nature of this change. `pip` is ran with `--upgrade` and `--upgrade-strategy eager` so it will try to upgrade the packages passed to the command and their dependencies to the latest version. That list of packages has become a lot longer and many of them are pinned to an exact version, so finding a compatible version of a certain dependency has become more difficult. I don't think there is a lot we can do about this.

Reply:
There is currently some uncertainty about the exact behavior. I am going to run some manual tests to verify that this is stable:
- install core in venv1, use venv2 as compiler venv, run `inmanta project update`, verify that any updated dependencies of core are shadowed in venv2 rather than reinstalled in venv1.

(part FYI, part note to self because I need to focus on another review first)

Reply:
Seems to be fine.